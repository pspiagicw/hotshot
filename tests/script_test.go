package tests

import (
	"os/exec"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

type scriptTest struct {
	name string
	file string
}

func checkFile(t *testing.T, filename string) {
	cmd := exec.Command("../hotshot", filename)
	output, err := cmd.Output()
	if err != nil {
		t.Errorf("Error running interpreter! %v", err)
	}
	snaps.MatchSnapshot(t, string(output))
}

func TestScripts(t *testing.T) {
	t.Skip()
	tt := []scriptTest{
		{
			name: "Hello World",
			file: "files/hello-world.ht",
		},
		{
			name: "Data Types",
			file: "files/data-types.ht",
		},
		{
			name: "Arithmetic Types",
			file: "files/arithmetic.ht",
		},
		{
			name: "Booleans",
			file: "files/booleans.ht",
		},
	}

	for _, test := range tt {
		t.Run(test.name, func(t *testing.T) {
			checkFile(t, test.file)
		})
	}

}
