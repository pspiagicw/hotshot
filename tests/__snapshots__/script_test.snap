
[TestScripts/Hello_World - 1]
 Hello, World 

---

[TestScripts/Data_Types - 1]
 2 
 this is a string 
 true 
 false 

---

[TestScripts/Arithmetic_Types - 1]
 3 
 17 
 1 
 14 
 ERROR: MULTIPLY function expects Integer, found STRING 
 ERROR: ADD function expects Integer, found STRING 
 ERROR: MINUS function expects Integer, found STRING 
 ERROR: DIVIDE function expects Integer, found STRING 
 ERROR: MULTIPLY function expects Integer, found BOOLEAN 
 ERROR: ADD function expects Integer, found BOOLEAN 
 ERROR: MINUS function expects Integer, found BOOLEAN 
 ERROR: DIVIDE function expects Integer, found BOOLEAN 

---

[TestScripts/Booleans - 1]
 true 
 false 
 true 
 false 
 false 
 true 
 false 
 false 
 false 
 true 
 true 
 true 
 ERROR: NOT function expects Boolean, found STRING 
 ERROR: NOT function expects Boolean, found STRING 
 ERROR: AND function expects Boolean, found STRING 
 ERROR: AND function expects Boolean, found STRING 
 ERROR: AND function expects Boolean, found STRING 
 ERROR: AND function expects Boolean, found STRING 
 ERROR: OR function expects Boolean, found STRING 
 ERROR: OR function expects Boolean, found STRING 
 ERROR: OR function expects Boolean, found STRING 
 ERROR: OR function expects Boolean, found STRING 
 true 
 false 
 false 
 false 
 false 
 true 
 ERROR: GT function expects Integer, found STRING 
 ERROR: LT function expects Integer, found INTEGER 

---
